src/aklite_client_ext.cc

```diff
@@ -276,3 +277,51 @@ bool AkliteClientExt::RebootIfRequired() {
 
   return false;
 }
+
+InstallResult AkliteClientExt::Rollback(const LocalUpdateSource* local_update_source, int version,
+                                        const std::string& target_name) {
+  auto pending_target = Target::toTufTarget(client_->getPendingTarget());
+  auto current = GetCurrent();
+
+  auto storage = INvStorage::newStorage(client_->config.storage, false);
+  auto update_mode =
+      pending_target.IsUnknown() ? InstalledVersionUpdateMode::kBadTarget : InstalledVersionUpdateMode::kNone;
+
+  // Set current target as failing
+  storage->saveInstalledVersion("", Target::fromTufTarget(current), update_mode);
+
+  // Get rollback target
+  TufTarget rollback_target;
+  if (version == -1 && target_name.empty()) {
+    rollback_target = Target::toTufTarget(client_->getRollbackTarget(true));
+    if (rollback_target.IsUnknown() || rollback_target.Name() == current.Name()) {
```

Comment:
I am not sure if this `rollback_target.Name() == current.Name()` works if a new ostree version is installed and a device is not installed yet.